src/app/views/settings/Settings.tsx

```diff
@@ -0,0 +1,155 @@
+import {
+  ChoiceGroup, Dialog, DialogType, IconButton
+} from 'office-ui-fabric-react';
+import React, { Component } from 'react';
+import { injectIntl } from 'react-intl';
+import { connect } from 'react-redux';
+import { bindActionCreators, Dispatch } from 'redux';
+import { loadGETheme } from '../../../themes';
+import { IThemeChangedMessage } from '../../../types/query-runner';
+import { ISettingsProps } from '../../../types/settings';
+import * as authActionCreators from '../../services/actions/auth-action-creators';
+import * as themeAtionCreators from '../../services/actions/theme-action-creator';
+
+class Settings extends Component<ISettingsProps, any> {
+  constructor(props: ISettingsProps) {
+    super(props);
+    this.state = {
+      hideThemeChooserDialog: true,
+    };
+  }
+
+  public toggleThemeChooserDialogState = () => {
+    this.setState({ hideThemeChooserDialog: !this.state.hideThemeChooserDialog });
+  }
+
+  public handleSignOut = () => {
+    this.props.actions!.signOut();
+  }
+
+  public handleChangeTheme = (event: any, option: any) => {
+    const newTheme: IThemeChangedMessage['theme'] = option.key;
+    this.props.actions!.changeTheme(newTheme);
+    loadGETheme(newTheme);
+  }
+
+  public render() {
+
+    const {
+      intl: { messages },
+      authenticated,
+      appTheme
+    }: any = this.props;
+
+    const { hideThemeChooserDialog } = this.state;
+
+    const items: any = [
+      {
+        key: 'office-dev-program',
+        text: messages['Office Dev Program'],
+        href: 'https://developer.microsoft.com/en-us/office/dev-program',
+        target: '_blank',
+        iconProps: {
+          iconName: 'CommandPrompt',
+        },
+      },
+      {
+        key: 'change-theme',
+        text: messages['Change theme'],
+        iconProps: {
+          iconName: 'Color',
+        },
+        onClick: () => this.toggleThemeChooserDialogState(),
+      }
+    ];
+
+    if (authenticated) {
+      items.push({
+        key: 'sign-out',
+        text: messages['sign out'],
+        iconProps: {
+          iconName: 'SignOut',
+        },
+        onClick: () => this.handleSignOut(),
+      });
+    }
+
+    const menuProperties = {
+      shouldFocusOnMount: true,
+      alignTargetEdge: true,
+      items
+    };
+
+    return (
+      <>
```

Comment:
What do the empty tags do?

Reply:
They create fragments so when we do not need to nest elements in a div element